src/App.js

```diff
@@ -1,25 +1,60 @@
 import React from 'react';
-
+import Clock from './Clock';
+import { Button } from './Button';
 import './App.scss';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: 1,
+  };
+
+  changeName = () => {
+    this.setState({
+      clockName: Math.ceil(Math.random() * (10 - 1)),
+    });
+  }
+
+  visibilityOn = () => {
+    this.setState(state => ({
+      isClockVisible: true,
+    }));
+  };
+
+  visibilityOff = () => {
+    this.setState(state => ({
+      isClockVisible: false,
+    }));
+  };
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+  render() {
+    const clockVisibility = this.state.isClockVisible
+      ? <Clock clockName={this.state.clockName} />
+      : 'Time to sleep';
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+    return (
+      <>
+        <h1>React clock</h1>
+        <div className="time">
+          Current time:
+          {' '}
+          {clockVisibility}
+        </div>
+        <Button
+          callback={this.visibilityOn}
```

Comment:
```suggestion
          onClick={this.visibilityOn}
```

В загальному контексті не до кінця зрозуміло, за що відповідає callback, а так підвищиться читабельність